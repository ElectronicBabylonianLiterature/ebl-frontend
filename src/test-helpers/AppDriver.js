import React from 'react'
import { fireEvent, render, waitForElement } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import App from 'App'
import Auth from 'auth/Auth'
import WordRepository from 'dictionary/WordRepository'
import FragmentRepository from 'fragmentarium/FragmentRepository'
import ImageRepository from 'fragmentarium/ImageRepository'
import FragmentService from 'fragmentarium/FragmentService'
import WordService from 'dictionary/WordService'
import TextService from 'corpus/TextService'
import Session from 'auth/Session'
import BibliographyRepository from 'bibliography/BibliographyRepository'
import BibliographyService from 'bibliography/BibliographyService'
import { defaultErrorReporter } from 'ErrorReporterContext'
import createAuth0Config from 'auth/createAuth0Config'

function createApp(api, sessionStore) {
  const auth0Config = createAuth0Config()
  const auth = new Auth(sessionStore, defaultErrorReporter, auth0Config)
  const wordRepository = new WordRepository(api)
  const fragmentRepository = new FragmentRepository(api)
  const imageRepository = new ImageRepository(api)
  const bibliographyRepository = new BibliographyRepository(api)
  const wordService = new WordService(wordRepository)
  const bibliographyService = new BibliographyService(bibliographyRepository)
  const fragmentService = new FragmentService(
    fragmentRepository,
    imageRepository,
    bibliographyService
  )
  const textService = new TextService(api)

  return (
    <App
      auth={auth}
      wordService={wordService}
      fragmentService={fragmentService}
      bibliographyService={bibliographyService}
      textService={textService}
    />
  )
}

export default class AppDriver {
  initialEntries = []
  element = null
  session = null

  constructor(api) {
    this.api = api
  }

  withPath(path) {
    this.initialEntries = [path]
    return this
  }

  withSession() {
    this.session = new Session(
      'accessToken',
      'idToken',
      Number.MAX_SAFE_INTEGER,
      ['write:texts']
    )
    return this
  }

  render() {
    const fakeSessionStore = {
      setSession: session => {
        this.session = session
      },
      clearSession: () => {
        this.session = null
      },
      getSession: () => this.session || new Session('', '', 0, [])
    }

    this.element = render(
      <MemoryRouter initialEntries={this.initialEntries}>
        {createApp(this.api, fakeSessionStore)}
      </MemoryRouter>
    )

    return this
  }

  async waitForText(text) {
    await waitForElement(() => this.element.getByText(text))
  }

  expectTextContent(text) {
    expect(this.element.container).toHaveTextContent(text)
  }

  expectNotInContent(text) {
    expect(this.element.queryByText(text)).toBeNull()
  }

  expectLink(text, expectedHref) {
    expect(this.element.getByText(text)).toHaveAttribute('href', expectedHref)
  }

  expectBreadcrumbs(crumbs) {
    this.expectTextContent(crumbs.join(''))
  }

  expectInputElement(label, expectedValue) {
    expect(this.element.getByLabelText(label).value).toEqual(
      String(expectedValue)
    )
  }

  changeValueByLabel(label, newValue) {
    const input = this.element.getByLabelText(label)
    fireEvent.change(input, { target: { value: newValue } })
  }

  click(text, n = 0) {
    const clickable = this.element.getAllByText(text)[n]
    fireEvent.click(clickable)
    // await wait()
  }
}
