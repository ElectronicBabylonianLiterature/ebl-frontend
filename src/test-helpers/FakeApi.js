import Promise from 'bluebird'

export default class FakeApi {
  gets = []
  client = {
    fetchJson: (path, authenticate) => {
      const expectation = this.gets.find(entry => entry.path === path && entry.authenticate === authenticate)
      return expectation
        ? Promise.resolve(expectation.response)
        : Promise.reject(new Error(`Unexpected ${authenticate ? 'authenticated' : 'not-authenticated'} fetchJson: ${path}`))
    },

    fetchBlob: (path, authenticate) => {
      return Promise.reject(new Error(`Unexpected ${authenticate ? 'authenticated' : 'not-authenticated'} fetchBlob: ${path}`))
    }
  }

  expectText (text) {
    this.gets.push({
      path: `/texts/${text.category}/${text.index}`,
      authenticate: true,
      response: text
    })
    return this
  }

  expectStatistics (statistics) {
    this.gets.push({
      path: `/statistics`,
      authenticate: false,
      response: statistics
    })
    return this
  }
}
