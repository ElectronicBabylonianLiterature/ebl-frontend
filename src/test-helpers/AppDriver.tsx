import React from 'react'
import {
  fireEvent,
  render,
  RenderResult,
  act,
  Matcher,
  within
} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import _ from 'lodash'
import App from 'App'
import WordRepository from 'dictionary/infrastructure/WordRepository'
import FragmentRepository from 'fragmentarium/infrastructure/FragmentRepository'
import ApiImageRepository from 'fragmentarium/infrastructure/ImageRepository'
import FragmentService from 'fragmentarium/application/FragmentService'
import WordService from 'dictionary/application/WordService'
import TextService from 'corpus/TextService'
import MemorySession, { Session, guestSession } from 'auth/Session'
import BibliographyRepository from 'bibliography/infrastructure/BibliographyRepository'
import BibliographyService from 'bibliography/application/BibliographyService'
import FragmentSearchService from 'fragmentarium/application/FragmentSearchService'
import { Promise } from 'bluebird'
import { submitForm } from 'test-helpers/utils'
import { eblNameProperty, AuthenticationContext } from 'auth/Auth'

function createApp(api): JSX.Element {
  const wordRepository = new WordRepository(api)
  const fragmentRepository = new FragmentRepository(api)
  const imageRepository = new ApiImageRepository(api)
  const bibliographyRepository = new BibliographyRepository(api)
  const wordService = new WordService(wordRepository)
  const bibliographyService = new BibliographyService(bibliographyRepository)
  const fragmentService = new FragmentService(
    fragmentRepository,
    imageRepository,
    wordRepository,
    bibliographyService
  )
  const fragmentSearchService = new FragmentSearchService(fragmentRepository)
  const textService = new TextService(api)

  return (
    <App
      wordService={wordService}
      fragmentService={fragmentService}
      fragmentSearchService={fragmentSearchService}
      bibliographyService={bibliographyService}
      textService={textService}
    />
  )
}

export default class AppDriver {
  private readonly api
  private initialEntries: string[] = []
  private element: RenderResult | null = null
  private session: Session | null = null

  constructor(api) {
    this.api = api
  }

  getElement(): RenderResult {
    if (this.element) {
      return this.element
    } else {
      throw new Error('getElement called before render.')
    }
  }

  withPath(path): AppDriver {
    this.initialEntries = [path]
    return this
  }

  withSession(): AppDriver {
    this.session = new MemorySession([
      'write:texts',
      'read:fragments',
      'annotate:fragments',
      'read:words'
    ])
    return this
  }

  async render(): Promise<AppDriver> {
    await act(async () => {
      this.element = render(
        <MemoryRouter initialEntries={this.initialEntries}>
          <AuthenticationContext.Provider
            value={{
              login: _.noop,
              logout: _.noop,
              getSession: (): Session => this.session ?? guestSession,
              isAuthenticated: (): boolean => this.session !== null,
              getAccessToken(): Promise<string> {
                throw new Error('Not implemented')
              },
              getUser(): { [eblNameProperty]: string } {
                return { [eblNameProperty]: 'Test' }
              }
            }}
          >
            {createApp(this.api)}
          </AuthenticationContext.Provider>
        </MemoryRouter>
      )
    })

    return this
  }

  async waitForText(text: Matcher): Promise<void> {
    await this.getElement().findByText(text)
  }

  expectTextContent(text: string | RegExp): void {
    expect(this.getElement().container).toHaveTextContent(text)
  }

  expectNotInContent(text: Matcher): void {
    expect(this.getElement().queryByText(text)).toBeNull()
  }

  expectLink(text: Matcher, expectedHref: string): void {
    expect(this.getElement().getByText(text)).toHaveAttribute(
      'href',
      expectedHref
    )
  }

  expectBreadcrumbs(crumbs: readonly string[]): void {
    this.expectTextContent(crumbs.join(''))
  }

  expectBreadcrumb(crumb: string, link: string): void {
    expect(
      within(this.getElement().getByLabelText('breadcrumb')).getByText(crumb)
    ).toHaveAttribute('href', link)
  }

  expectInputElement(label: Matcher, expectedValue: unknown): void {
    expect(
      (this.getElement().getByLabelText(label) as HTMLInputElement).value
    ).toEqual(String(expectedValue))
  }

  changeValueByLabel(label: Matcher, newValue): void {
    const input = this.getElement().getByLabelText(label)
    act(() => {
      fireEvent.change(input, { target: { value: newValue } })
    })
  }

  async submitForm(): Promise<void> {
    await submitForm(this.getElement())
  }

  async click(text: Matcher, n = 0): Promise<void> {
    const clickable = this.getElement().getAllByText(text)[n]
    await act(async () => {
      fireEvent.click(clickable)
    })
  }
}
