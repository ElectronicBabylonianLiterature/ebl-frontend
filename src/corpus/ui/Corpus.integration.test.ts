import _ from 'lodash'
import AppDriver from 'test-support/AppDriver'
import FakeApi from 'test-support/FakeApi'

const textsDto = [
  {
    genre: 'D',
    category: 3,
    index: 1,
    name: 'Divination Third Category',
    numberOfVerses: 674,
    approximateVerses: true,
    chapters: [],
  },
  {
    genre: 'Lex',
    category: 1,
    index: 1,
    name: 'Lexicography First Category',
    numberOfVerses: 3,
    approximateVerses: false,
    chapters: [],
  },
  {
    genre: 'L',
    category: 1,
    index: 2,
    name: 'First Category 2',
    numberOfVerses: 99,
    approximateVerses: false,
    chapters: [],
  },
  {
    genre: 'L',
    category: 2,
    index: 1,
    name: 'Second Category',
    numberOfVerses: 2,
    approximateVerses: false,
    chapters: [],
  },
  {
    genre: 'L',
    category: 3,
    index: 1,
    name: 'Third Category',
    numberOfVerses: 45,
    approximateVerses: true,
    chapters: [],
  },
  {
    genre: 'L',
    category: 1,
    index: 1,
    name: 'First Category 1',
    numberOfVerses: 99,
    approximateVerses: false,
    chapters: [],
  },
  {
    genre: 'L',
    category: 4,
    index: 1,
    name: 'Hidden Text',
    numberOfVerses: 1,
    approximateVerses: false,
    chapters: [],
  },
]

let fakeApi: FakeApi
let appDriver: AppDriver

afterEach(() => {
  fakeApi.verifyExpectations()
})

beforeEach(() => {
  fakeApi = new FakeApi()
  fakeApi.expectTexts(textsDto)
  fakeApi.allowImage('LibraryCropped.svg')
  appDriver = new AppDriver(fakeApi.client).withPath('/corpus')
})

test('With session', async () => {
  await appDriver.withSession().render()
  await appDriver.waitForText(RegExp(_.escapeRegExp('Narrative Poetry')))
  expect(appDriver.getElement().container).toMatchSnapshot()
})

test('Without session', async () => {
  await appDriver.render()
  await appDriver.waitForText(RegExp(_.escapeRegExp('Narrative Poetry')))
  expect(appDriver.getElement().container).toMatchSnapshot()
})
