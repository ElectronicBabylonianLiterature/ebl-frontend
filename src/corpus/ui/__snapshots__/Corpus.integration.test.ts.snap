// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With session 1`] = `
<div>
  <div
    class="main-body"
  >
    <header
      class="Header"
    >
      <nav
        class="navbar navbar-expand-md navbar-light"
      >
        <div
          class="container"
        >
          <a
            class="active navbar-brand"
            href="/"
            title="electronic Babylonian Library (eBL)"
          >
            <h1
              class="Header__title"
            >
              <span
                class="Header__title-main"
              >
                electronic
                <br />
                <span
                  class="Header__title-main__caps"
                >
                  B
                </span>
                abylonian
                <br />
                <span
                  class="Header__title-main__caps"
                >
                  L
                </span>
                ibrary
              </span>
              <small
                class="Header__title-abbreviation"
              >
                e
                <small
                  class="Header__title-abbreviation__caps"
                >
                  bl
                </small>
              </small>
            </h1>
          </a>
          <span
            class="navbar-brand"
          >
            <div
              class="Header__logo-container container"
            >
              <a
                href="https://www.lmu.de"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  class="Header__lmu-logo img-fluid"
                  src="LMU_Logo.svg"
                />
              </a>
              <a
                href="https://badw.de/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  class="Header__badw-logo img-fluid"
                  src="BAdW_Logo.svg"
                />
              </a>
            </div>
          </span>
          <button
            aria-controls="Header-1"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="Header-1"
          >
            <div
              class="d-flex justify-content-between"
              id="navbar-container"
            >
              <div
                id="menu-lines"
              >
                <div
                  class="mx-auto navbar-nav"
                >
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/signs"
                      href="/signs"
                    >
                      Signs
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/dictionary"
                      href="/dictionary"
                    >
                      Dictionary
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="active nav-link"
                      data-rb-event-key="/corpus"
                      href="/corpus"
                    >
                      Corpus
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/library"
                      href="/library"
                    >
                      Library
                    </a>
                  </div>
                </div>
                <div
                  class="mx-auto navbar-nav"
                >
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/about"
                      href="/about"
                    >
                      About
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/bibliography"
                      href="/bibliography"
                    >
                      Bibliography
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/tools"
                      href="/tools"
                    >
                      Tools
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/projects"
                      href="/projects"
                    >
                      Projects
                    </a>
                  </div>
                </div>
              </div>
              <span
                class="navbar-text"
                id="user"
              >
                <button
                  class="btn btn-outline-secondary btn-sm"
                  type="button"
                >
                  Logout 
                  Test
                </button>
              </span>
            </div>
          </div>
        </div>
      </nav>
    </header>
    <main
      class="main"
    >
      <div
        class="main__content"
      >
        <header
          class="main__header"
        >
          <nav
            aria-label="breadcrumb"
            class="main__breadcrumbs"
          >
            <ol
              class="breadcrumb"
            >
              <li
                class="breadcrumb-item active"
              >
                <a
                  href="/"
                >
                  eBL
                </a>
              </li>
              <li
                aria-current="page"
                class="breadcrumb-item active"
              >
                Corpus
              </li>
            </ol>
          </nav>
          <h2
            class="main__heading"
          >
            Corpus
          </h2>
          <div
            class="main__toolbar btn-toolbar"
            role="toolbar"
          />
        </header>
        <div
          class="container-fluid"
        >
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <nav
                class="nav nav-tabs"
                role="tablist"
              >
                <a
                  aria-controls="CorpusTab-2-tabpane-L"
                  aria-selected="true"
                  class="nav-item nav-link active"
                  data-rb-event-key="L"
                  href="#"
                  id="CorpusTab-2-tab-L"
                  role="tab"
                >
                  Literature
                </a>
                <a
                  aria-controls="CorpusTab-2-tabpane-D"
                  aria-selected="false"
                  class="nav-item nav-link"
                  data-rb-event-key="D"
                  href="#"
                  id="CorpusTab-2-tab-D"
                  role="tab"
                >
                  Divination
                </a>
                <a
                  aria-controls="CorpusTab-2-tabpane-Med"
                  aria-selected="false"
                  class="nav-item nav-link"
                  data-rb-event-key="Med"
                  href="#"
                  id="CorpusTab-2-tab-Med"
                  role="tab"
                >
                  Medicine
                </a>
                <a
                  aria-controls="CorpusTab-2-tabpane-Mag"
                  aria-selected="false"
                  class="nav-item nav-link"
                  data-rb-event-key="Mag"
                  href="#"
                  id="CorpusTab-2-tab-Mag"
                  role="tab"
                >
                  Magic
                </a>
              </nav>
              <div
                class="tab-content"
              >
                <div
                  aria-hidden="false"
                  aria-labelledby="CorpusTab-2-tab-L"
                  class="fade tab-pane active show"
                  id="CorpusTab-2-tabpane-L"
                  role="tabpanel"
                >
                  <section>
                    <h3 />
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      I. Narrative Poetry
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/L/1/1"
                          >
                            1
                            . 
                            First Category 1
                          </a>
                           
                          — 
                          
                          99
                           vv.
                        </div>
                      </li>
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/L/1/2"
                          >
                            2
                            . 
                            First Category 2
                          </a>
                           
                          — 
                          
                          99
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                  <section>
                    <h3>
                      II. Monologue and dialogue literature
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/L/2/1"
                          >
                            1
                            . 
                            Second Category
                          </a>
                           
                          — 
                          
                          2
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                  <section>
                    <h3>
                      III. Literary Hymns and Prayers
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/L/3/1"
                          >
                            1
                            . 
                            Third Category
                          </a>
                           
                          — 
                          ±
                          45
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                </div>
                <div
                  aria-hidden="true"
                  aria-labelledby="CorpusTab-2-tab-D"
                  class="fade tab-pane"
                  id="CorpusTab-2-tabpane-D"
                  role="tabpanel"
                >
                  <section>
                    <h3 />
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      I. Celestial Divination (
                      <em>
                        Enūma Anu Enlil
                      </em>
                      )
                    </h3>
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      II. Terrestrial Divination (
                      <em>
                        Šumma Ālu
                      </em>
                      )
                    </h3>
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      III. Extispicy (
                      <em>
                        Bārûtu
                      </em>
                      )
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/D/3/1"
                          >
                            1
                            . 
                            Divination Third Category
                          </a>
                           
                          — 
                          ±
                          674
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                </div>
                <div
                  aria-hidden="true"
                  aria-labelledby="CorpusTab-2-tab-Med"
                  class="fade tab-pane"
                  id="CorpusTab-2-tabpane-Med"
                  role="tabpanel"
                >
                  <section>
                    <h3 />
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      I. Nineveh Medical Encyclopaedia
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/Med/1/1"
                          >
                            1
                            . 
                            Medicine First Category
                          </a>
                           
                          — 
                          
                          3
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                </div>
                <div
                  aria-hidden="true"
                  aria-labelledby="CorpusTab-2-tab-Mag"
                  class="fade tab-pane"
                  id="CorpusTab-2-tabpane-Mag"
                  role="tabpanel"
                >
                  <section>
                    <h3 />
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      I. Anti-witchcraft
                    </h3>
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      II. “Hand-lifting” Prayers (
                      <em>
                        šuʾilas
                      </em>
                      )
                    </h3>
                    <ol
                      class="container-fluid"
                    />
                  </section>
                </div>
              </div>
            </div>
            <div
              class="col-md-6"
            >
              <img
                alt="LibraryCropped.svg"
                class="img-fluid"
                src="http://example.com/images/LibraryCropped.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="main-footer"
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/impressum"
          >
            Impressum
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/datenschutz"
          >
            Datenschutz
          </a>
        </li>
      </ol>
    </footer>
  </div>
</div>
`;

exports[`Without session 1`] = `
<div>
  <div
    class="main-body"
  >
    <header
      class="Header"
    >
      <nav
        class="navbar navbar-expand-md navbar-light"
      >
        <div
          class="container"
        >
          <a
            class="active navbar-brand"
            href="/"
            title="electronic Babylonian Library (eBL)"
          >
            <h1
              class="Header__title"
            >
              <span
                class="Header__title-main"
              >
                electronic
                <br />
                <span
                  class="Header__title-main__caps"
                >
                  B
                </span>
                abylonian
                <br />
                <span
                  class="Header__title-main__caps"
                >
                  L
                </span>
                ibrary
              </span>
              <small
                class="Header__title-abbreviation"
              >
                e
                <small
                  class="Header__title-abbreviation__caps"
                >
                  bl
                </small>
              </small>
            </h1>
          </a>
          <span
            class="navbar-brand"
          >
            <div
              class="Header__logo-container container"
            >
              <a
                href="https://www.lmu.de"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  class="Header__lmu-logo img-fluid"
                  src="LMU_Logo.svg"
                />
              </a>
              <a
                href="https://badw.de/"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  class="Header__badw-logo img-fluid"
                  src="BAdW_Logo.svg"
                />
              </a>
            </div>
          </span>
          <button
            aria-controls="Header-4"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="Header-4"
          >
            <div
              class="d-flex justify-content-between"
              id="navbar-container"
            >
              <div
                id="menu-lines"
              >
                <div
                  class="mx-auto navbar-nav"
                >
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/signs"
                      href="/signs"
                    >
                      Signs
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/dictionary"
                      href="/dictionary"
                    >
                      Dictionary
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="active nav-link"
                      data-rb-event-key="/corpus"
                      href="/corpus"
                    >
                      Corpus
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/library"
                      href="/library"
                    >
                      Library
                    </a>
                  </div>
                </div>
                <div
                  class="mx-auto navbar-nav"
                >
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/about"
                      href="/about"
                    >
                      About
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/bibliography"
                      href="/bibliography"
                    >
                      Bibliography
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/tools"
                      href="/tools"
                    >
                      Tools
                    </a>
                  </div>
                  <div
                    class="nav-item"
                  >
                    <a
                      class="nav-link"
                      data-rb-event-key="/projects"
                      href="/projects"
                    >
                      Projects
                    </a>
                  </div>
                </div>
              </div>
              <span
                class="navbar-text"
                id="user"
              >
                <button
                  class="btn btn-outline-secondary btn-sm"
                  type="button"
                >
                  Login
                </button>
              </span>
            </div>
          </div>
        </div>
      </nav>
    </header>
    <main
      class="main"
    >
      <div
        class="main__content"
      >
        <header
          class="main__header"
        >
          <nav
            aria-label="breadcrumb"
            class="main__breadcrumbs"
          >
            <ol
              class="breadcrumb"
            >
              <li
                class="breadcrumb-item active"
              >
                <a
                  href="/"
                >
                  eBL
                </a>
              </li>
              <li
                aria-current="page"
                class="breadcrumb-item active"
              >
                Corpus
              </li>
            </ol>
          </nav>
          <h2
            class="main__heading"
          >
            Corpus
          </h2>
          <div
            class="main__toolbar btn-toolbar"
            role="toolbar"
          />
        </header>
        <div
          class="container-fluid"
        >
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <nav
                class="nav nav-tabs"
                role="tablist"
              >
                <a
                  aria-controls="CorpusTab-5-tabpane-L"
                  aria-selected="true"
                  class="nav-item nav-link active"
                  data-rb-event-key="L"
                  href="#"
                  id="CorpusTab-5-tab-L"
                  role="tab"
                >
                  Literature
                </a>
                <a
                  aria-controls="CorpusTab-5-tabpane-D"
                  aria-selected="false"
                  class="nav-item nav-link"
                  data-rb-event-key="D"
                  href="#"
                  id="CorpusTab-5-tab-D"
                  role="tab"
                >
                  Divination
                </a>
                <a
                  aria-controls="CorpusTab-5-tabpane-Med"
                  aria-selected="false"
                  class="nav-item nav-link"
                  data-rb-event-key="Med"
                  href="#"
                  id="CorpusTab-5-tab-Med"
                  role="tab"
                >
                  Medicine
                </a>
                <a
                  aria-controls="CorpusTab-5-tabpane-Mag"
                  aria-selected="false"
                  class="nav-item nav-link"
                  data-rb-event-key="Mag"
                  href="#"
                  id="CorpusTab-5-tab-Mag"
                  role="tab"
                >
                  Magic
                </a>
              </nav>
              <div
                class="tab-content"
              >
                <div
                  aria-hidden="false"
                  aria-labelledby="CorpusTab-5-tab-L"
                  class="fade tab-pane active show"
                  id="CorpusTab-5-tabpane-L"
                  role="tabpanel"
                >
                  <section>
                    <h3 />
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      I. Narrative Poetry
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/L/1/1"
                          >
                            1
                            . 
                            First Category 1
                          </a>
                           
                          — 
                          
                          99
                           vv.
                        </div>
                      </li>
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/L/1/2"
                          >
                            2
                            . 
                            First Category 2
                          </a>
                           
                          — 
                          
                          99
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                  <section>
                    <h3>
                      II. Monologue and dialogue literature
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/L/2/1"
                          >
                            1
                            . 
                            Second Category
                          </a>
                           
                          — 
                          
                          2
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                  <section>
                    <h3>
                      III. Literary Hymns and Prayers
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/L/3/1"
                          >
                            1
                            . 
                            Third Category
                          </a>
                           
                          — 
                          ±
                          45
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                </div>
                <div
                  aria-hidden="true"
                  aria-labelledby="CorpusTab-5-tab-D"
                  class="fade tab-pane"
                  id="CorpusTab-5-tabpane-D"
                  role="tabpanel"
                >
                  <section>
                    <h3 />
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      I. Celestial Divination (
                      <em>
                        Enūma Anu Enlil
                      </em>
                      )
                    </h3>
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      II. Terrestrial Divination (
                      <em>
                        Šumma Ālu
                      </em>
                      )
                    </h3>
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      III. Extispicy (
                      <em>
                        Bārûtu
                      </em>
                      )
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/D/3/1"
                          >
                            1
                            . 
                            Divination Third Category
                          </a>
                           
                          — 
                          ±
                          674
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                </div>
                <div
                  aria-hidden="true"
                  aria-labelledby="CorpusTab-5-tab-Med"
                  class="fade tab-pane"
                  id="CorpusTab-5-tabpane-Med"
                  role="tabpanel"
                >
                  <section>
                    <h3 />
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      I. Nineveh Medical Encyclopaedia
                    </h3>
                    <ol
                      class="container-fluid"
                    >
                      <li
                        class="row"
                      >
                        <div
                          class="col"
                        >
                          <a
                            href="/corpus/Med/1/1"
                          >
                            1
                            . 
                            Medicine First Category
                          </a>
                           
                          — 
                          
                          3
                           vv.
                        </div>
                      </li>
                    </ol>
                  </section>
                </div>
                <div
                  aria-hidden="true"
                  aria-labelledby="CorpusTab-5-tab-Mag"
                  class="fade tab-pane"
                  id="CorpusTab-5-tabpane-Mag"
                  role="tabpanel"
                >
                  <section>
                    <h3 />
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      I. Anti-witchcraft
                    </h3>
                    <ol
                      class="container-fluid"
                    />
                  </section>
                  <section>
                    <h3>
                      II. “Hand-lifting” Prayers (
                      <em>
                        šuʾilas
                      </em>
                      )
                    </h3>
                    <ol
                      class="container-fluid"
                    />
                  </section>
                </div>
              </div>
            </div>
            <div
              class="col-md-6"
            >
              <img
                alt="LibraryCropped.svg"
                class="img-fluid"
                src="http://example.com/images/LibraryCropped.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="main-footer"
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/impressum"
          >
            Impressum
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/datenschutz"
          >
            Datenschutz
          </a>
        </li>
      </ol>
    </footer>
  </div>
</div>
`;
